packages/patrol_cli/lib/src/common/artifacts_repository.dart

```diff
@@ -3,30 +3,33 @@ import 'package:file/file.dart';
 import 'package:http/http.dart' as http;
 import 'package:path/path.dart' show join, dirname;
 import 'package:patrol_cli/src/common/constants.dart' show version;
-import 'package:patrol_cli/src/top_level_flags.dart';
 import 'package:platform/platform.dart';
 
 class ArtifactsRepository {
   ArtifactsRepository({
     required FileSystem fs,
-    required Platform platform,
-    required TopLevelFlags topLevelFlags,
+    required this.platform,
+    http.Client? httpClient,
+    ZipDecoder? zipDecoder,
   })  : _fs = fs,
-        _platform = platform,
-        _topLevelFlags = topLevelFlags {
+        _httpClient = httpClient ?? http.Client(),
+        _zipDecoder = zipDecoder ?? ZipDecoder(),
+        debug = false {
     _paths = _Paths(artifactPath);
   }
 
   static const artifactPathEnv = 'PATROL_CACHE';
 
   final FileSystem _fs;
-  final Platform _platform;
-  final TopLevelFlags _topLevelFlags;
+  Platform platform;
+  final http.Client _httpClient;
+  final ZipDecoder _zipDecoder;
+  bool debug;
```

Comment:
Not final?

Reply:
Nope, I set it after I know whether the user passed `--debug` (which is after the `ArtifactsRepository` is instantiated).